tfhe/src/shortint/engine/server_side/bitwise_op.rs

```diff
@@ -29,14 +29,20 @@ impl ShortintEngine {
         Ok(())
     }
 
+    // by convention smart operations take mut refs to their inputs, even if they do not modify them
+    #[allow(clippy::needless_pass_by_ref_mut)]
```

Comment:
that #[allow] should not be needed anymore,  in the function, message_extract_assign was added, which makes &mut ct_left and &mut ct_right needed